Data-Structures-and-Algorithms/SinglyLinkedList.py

```diff
@@ -0,0 +1,112 @@
+
+class node:
+	def __init__(self, data = None):
+		self.data = data
+		self.next = None
+
+class linked_list:
+	def __init__(self):
+		self.head = node()
+
+	# Adds new node containing 'data' to the end of the linked list.
+	def append(self, data):
+		new_node = node(data)
+		cur = self.head
+		while cur.next != None:
+			cur = cur.next
+		cur.next = new_node
+
+	# Returns the length (integer) of the linked list.
+	def length(self):
+		cur = self.head
+		total = 0
+		while cur.next != None:
+			total += 1
+			cur = cur.next
+		return total 
+
+	# Prints out the linked list in traditional Python list format. 
+	def display(self):
+		elems = []
+		cur_node = self.head
+		while cur_node.next != None:
+			cur_node = cur_node.next
+			elems.append(cur_node.data)
+		print(elems)
```

Comment:
If you are printing elements here then why storing them in the list?

Reply:
to print them as a list, if I directly print the elements, then won't it just print each element on different lines?

Reply:
learn how to print elements in the same line

Reply:
Well, I know that already, but never thought about using it here, thanks!